### PR TITLE
New Azuremonitorreceiver component

### PR DESCRIPTION
**Pull request by sriniketh923**
Description: Adds support for the [azuremonitor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/azuremonitorreceiver) receiver from the [opentelemetry contrib repo](https://github.com/open-telemetry/opentelemetry-collector-contrib).

Testing: components_test.go has been updated to reflect these changes. Local testing of the azuremonitorreceiver was also successful. We were able to scrape the azuremonitor metrics and send them to Splunk Observability.

Documentation: components.md has been updated to show the new component

Co-author: https://github.com/brendtg @brendtg

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### 🚀 New components 🚀
+
+- (Splunk) Add Azuremonitor receiver ([#4971](https://github.com/signalfx/splunk-otel-collector/pull/4971))
```
Comment (crobert-1) [inline CHANGELOG.md:7]:
```suggestion
- (Splunk) Add Azure Monitor receiver ([#4971](https://github.com/signalfx/splunk-otel-collector/pull/4971))
```
Nit

Comment (sriniketh923) [reply CHANGELOG.md]: Made the change

File: docs/components.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ The distribution offers support for the following components.
 | [awsecscontainermetrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/awsecscontainermetricsreceiver) | [beta]           |
 | [apache](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/apachereceiver)                                 | [alpha]          |
 | [azureeventhub](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/azureeventhubreceiver)                   | [alpha]          |
+| [azuremonitor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/azuremonitorreceiver)                     | [alpha]          |
```
Comment (crobert-1) [inline docs/components.md:20]:
```suggestion
| [azuremonitor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/azuremonitorreceiver)                     | [in development]          |
```
Stability is currently `development` from what I can tell. [Reference.](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/azuremonitorreceiver#azure-monitor-receiver)

Comment (sriniketh923) [reply docs/components.md]: Ohh yes just updated it.

**Review by crobert-1 — APPROVED** [review]
Looks good to me, thanks for your contribution here!

File: go.mod [hunk 3]
```diff
@@ -157,11 +158,13 @@ require (
 	dario.cat/mergo v1.0.0 // indirect
 	filippo.io/edwards25519 v1.1.0 // indirect
 	github.com/Azure/azure-amqp-common-go/v4 v4.2.0 // indirect
-	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.10.0 // indirect
-	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.5.1 // indirect
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.11.1 // indirect
+	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.5.2 // indirect
```
[the remark this answers is not in the document]
Comment (sriniketh923) [reply go.mod:162]: I cannot see that page. But I can bump azidentity and azcore to the latest versions

**Review by pjanotti — APPROVED** [review]

**Comment by sriniketh923** [comment]
The two integration tests seemed to have failed seem unrelated to this MR?

**Pull request merged**